client/src/components/ChannelFilter.tsx

```diff
@@ -0,0 +1,81 @@
+import Select, {SingleValue} from 'react-select';
+import {ChannelFilterProps} from '../lib/types';
+import React from 'react';
+import styled from 'styled-components';
+import {useCallback, useEffect, useState} from 'react';
+import {fetchData} from '../services/fetch';
+
+interface Channel {
+    id: string;
+    name: string;
+}
+
+interface Option {
+    label: string;
+    value: string;
+}
+
+const FilterDiv = styled.div`
+      min-width: 350px;
+      max-width: 50%;
+      flex-grow: 1;
+    `;
+
+const ChannelFilter = ({onSelected, selectedChannel}: ChannelFilterProps) => {
+    const allValue = {
+        label: 'Code Reviews For All Slack Channels',
+        value: 'all',
+    };
+    const [channelOptions, setChannelOptions] = useState([] as Option[]);
+    const [selectPlaceHolder, setSelectPlaceHolder] = useState(allValue.label);
+    const handleChannelSelectionChange = useCallback((data: SingleValue<Option>) => {
+        if (data?.label && data?.value) {
+            setSelectPlaceHolder(data.label);
+            onSelected(data.value);
+        }
+    }, [onSelected]);
+
+    useEffect(() => {
+        fetchData('/api/channels')
```

Comment:
This would be a good function for `useCallback` so that we do not execute a new API call on each render, but only when the deps change.

https://react.dev/reference/react/useCallback#preventing-an-effect-from-firing-too-often

Reply:
I think you misread the article, the fetchData is an external function, the reference function will never change when imported.  If you declare the fetchDation inside this component, then every time the component render, the function will get re-declare, hence cause a new reference to create.  Because there is no dependent reference to fetchData, it does not cause any re-render for this.  Since the function was declare outside of the component, that reference to the function will never get change.  So useCallback against the function declared outside of the component that has not external dependencies will not be any benefit.

The useEffect hook currently doesn't have any dependencies, so it only get trigger once on component mount.  On react 18, in dev mode, you will see this call twice because it did a unmount and re-mount.

Reply:
Making an API call on each renders seems like it could not be performant. Its not a blocker, but it might be nice to update this logic later to ensure the data is cached in some way to prevent excessive API calls.